### PR TITLE
Test updates in AC tests with row filtering and column masking

### PR DESCRIPTION
**Pull request by ksobolew**
The tests for access control with row filtering and column masking were incomplete with regards to update statements: the `TestColumnMask` already had tests for `INSERT` and `DELETE`, but `TestRowFilter` missed them, and both missed tests for `UPDATE`. This PR adds the missing pieces.

Includes a side-refactoring of exception message asserts.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/sql/query/TestColumnMask.java [hunk 1]
```diff
@@ -537,7 +537,24 @@ public void testDeleteWithColumnMasking()
...
+        accessControl.reset();
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "clerk",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "clerk"));
+        assertThatThrownBy(() -> assertions.query("UPDATE orders SET clerk = 'X'"))
+                .hasMessage("line 1:1: Updating a table with column masks is not supported");
+        assertThatThrownBy(() -> assertions.query("UPDATE orders SET orderkey = -orderkey"))
+                .hasMessage("line 1:1: Updating a table with column masks is not supported");
```
Comment (ksobolew) [inline core/trino-main/src/test/java/io/trino/sql/query/TestColumnMask.java:555]: `orderkey` here is not masked, but updates are still disallowed. Should updating only non-masked columns be allowed?

**Comment by findepi** [comment]
Should we run `BaseConnectorSmokeTest` against a setup with column masks?

i guess the expected result would be that it fails pretty miuch everywhere, but trying this would ensure ongoing test coverage, which is easy to not to add otherwise.
For example when working on UPDATE, or MERGE.

**Comment by kokosing** [comment]
> Should we run BaseConnectorSmokeTest against a setup with column masks?

I don't think so. Access controls and connectors are very independent. As no point to test access control when connector is tested.

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Merged, thanks.

**Comment by findepi** [comment]
> > Should we run BaseConnectorSmokeTest against a setup with column masks?
> 
> I don't think so. Access controls and connectors are very independent. As no point to test access control when connector is tested.

OTOH, masking is an "overlay" which affects engine behavior with respect to the connector.
Since it affects many different code paths -- SELECT, SHOW CREATE TABLE, INSERT, UPDATE, DELETE, ... -- it is desirable to run some suite of tests with masking enabled.

Today, BaseConnectorSmokeTest provides such suite of tests, but i am open to suggestions.

**Pull request merged**